Android with user-defined notification sound

rsulzenbacher opened the pull request:
When app in background always the default notification sound was playing.

I added following lines in GCMIntentService.java:

```
    NotificationCompat.Builder mBuilder =
        new NotificationCompat.Builder(context)
            .setDefaults(Notification.DEFAULT_ALL)  
            .setSmallIcon(context.getApplicationInfo().icon)
            .setWhen(System.currentTimeMillis())
            .setContentTitle(extras.getString("title"))
            .setTicker(extras.getString("title"))
            .setContentIntent(contentIntent);
```

after this lines i added:

```
    String soundName = extras.getString("soundname");
    if (soundName != null) {
        Uri soundUri = Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE + "://" + getPackageName() + "/raw/" + soundName.replace("-", "").replace(".mp3", ""));
        mBuilder.setDefaults(Notification.DEFAULT_LIGHTS | Notification.DEFAULT_VIBRATE);
        mBuilder.setSound(soundUri);            
    }
```

And i stored my mp3 files in res/raw folder. (without '-' in filename).

bobeasterday committed:
Reference Cordove 2.2.0 and obscure sender id from example

bobeasterday committed:
Cleanup documentation and sample layout

bobeasterday committed:
Cleanup file hierarchy and clarify it in manual installation section of readme

bobeasterday committed:
cleanup onLoad functionality

colene committed:
Update README.md

Fix apple link.

bobeasterday committed:
Merge pull request #1 from colene/patch-1

Update README.md

bobeasterday committed:
Add compatibility with 2.3.0

bobeasterday committed:
Add support for background notifications

bobeasterday committed:
Add v13 compatibility shim

bobeasterday committed:
Grab app icon without assuming package name

bobeasterday committed:
cleanup construction

bobeasterday committed:
Remove old cordova alias

bobeasterday committed:
Add flag for foreground vs. background for iOS

bobeasterday committed:
send entire payload to JS app

bobeasterday committed:
Clean up some unused imports

bobeasterday committed:
Update read me and examples to reflect status bar notifications changes

bobeasterday committed:
Surface custom key/value pairs to JS for iOS

bobeasterday committed:
Don't choke if no msgcnt or message values are passed in the notification

bobeasterday committed:
Dismiss notification from tray once it has been processed

bobeasterday committed:
Bump the plugin version

bobeasterday committed:
Always refresh the content Intent on new notifications.

bobeasterday committed:
and bump the version

bobeasterday committed:
Remove use of now private uniqueIdentifier accessor

bobeasterday committed:
bump version for uniqueidentifier fix

bobeasterday committed:
Add cold-start functionality

bobeasterday committed:
Update documentation to reflect new coldstart functionality

markeeftb committed:
Update README.md

You need to add android.permission.GET_TASKS. 
Without this permission, the application crashes when it receives PUSH NOTIFICATION.

bobeasterday committed:
Update PushPlugin for 2.5.0+

bobeasterday committed:
Get rid of deprecated call to getContext()

bobeasterday committed:
Merge pull request #3 from markeeftb/patch-1

Update README.md

bobeasterday committed:
Add plugman support

bobeasterday committed:
Add jasmin test spec

bobeasterday committed:
bump version after adding test spec

bobeasterday committed:
Fix the spec

filmaj committed:
fully qualified paths for ios.

bobeasterday committed:
Merge pull request #4 from filmaj/master

cordova-plugman support

Commit:
Updated to Cordova 2.7, fixed a number of bugs. Cleaned up some code.

* Put all the plugin code in the plugin namespace: com.plugin.gcm;
* Renamed to namespace GCM to gcm, namespaces should be lower case
according to the java style guidelines;
* Deleted the google gcm code and added gcm.jar instead;
* Added CordovaGCMBroadcastReceiver.java to enable GCMIntentService to
be namespaced in the plugin namespace instead of the client project
(which is is the default with google's GCMIntentService);
* Moved all java <--> javascript interop to PushPlugin;
* Moved all notification code to GCMIntentService (maybe this fits
better in its own class);
* Changed the way the plugin detects cold starts in order for it not to
crash when the back button is used to navigate out of the app;
* Changed the caching on the filesystem to a memory cache on a static
variable, which increases performance and simplifies code;
* Cut up some code into separate methods ;
* Added code comments;
* Updated the readme file.

Commit:
moved libs/gcm.jar to src/android/libs/gcm.jar for plugman compatibility

bobeasterday committed:
bump version to deliniate 1st Plugman version

bobeasterday committed:
Remove config file from spec for plugman compatibility

bobeasterday committed:
Accept either Boolean or string argiuments to register method for iOS

bobeasterday committed:
Merge pull request #5 from NetMatch/master

Updated to Cordova 2.7, fixed a number of bugs. Cleaned up some code.

bobeasterday committed:
Update documentation for pluginstall

Commit:
error callback for setApplicationIconBadgeNumber

bobeasterday committed:
Merge pull request #11 from Michael-vb/master

Fix for setApplicationIconBadgeNumber

bobeasterday committed:
surface payload when app is launched from incoming notification on iOS

bobeasterday committed:
Bump version

Commit:
[plugman] renaming GCM/ to gcm/ to make plugman work

Commit:
[1.3.2] upped version

bobeasterday committed:
Add description tag

Commit:
made code more readable and better :-)

goya committed:
[plugin.xml] add license tag

jdhiro committed:
Update plugin.xml

jdhiro committed:
Clean up and formatting

jdhiro committed:
Updating java imports to work with the new Cordova 3.0 API

jdhiro committed:
Updating iOS PushPlugin for ARC builds (cordova-cli 3.0 default configuration)

jdhiro committed:
Getting the plugin to register on iOS in a Cordova 3.0 environment

The signature of all the methods callable from JavaScript had to change
from:
-- (void)register:(NSMutableArray *)arguments
withDict:(NSMutableDictionary *)options;
to
+- (void)register:(CDVInvokedUrlCommand*)command;

The old format was officially obsoleted in in Cordova 3.0, so there is
no way around using the old one.

'arguments' and 'options' needed to use the new 'command' object.

With this new format the plugin result interface was also broken, so it
was changed to the new documented format.

This code is _not_ well tested at this point, but I wanted to get it up
for anyone having issues migrating to 3.0.

rjmunro committed:
Fix typo: successCallback should be errorCallback

Fix setApplicationIconBadgeNumber returning errors.

bobeasterday committed:
Restrict current version of plugin to PG < 3.0

bobeasterday committed:
remove trailing tab

bobeasterday committed:
Add author element

bobeasterday committed:
Escape < and > in engines element

bobeasterday committed:
Use canonical namespace prefix for plugin id

bobeasterday committed:
Make fuzzy version matching a little less fuzzy

bobeasterday committed:
Bump version to 1.3.5

rjmunro committed:
Eliminate need for GET_TASKS permission

* Don't check if we are in the foreground - that happens anyway in the
sendExtras() method (https://github.com/phonegap-build/PushPlugin/blob/master/src/android/com/plugin/gcm/PushPlugin.java#L111)
* Only display a message if the message has text to be displayed

Fixes #52 and undoes #3

bobeasterday committed:
Try with explicit version matching

bobeasterday committed:
bump version

plugin submission requires a version bump even if plugin was previously
deleted (bug?)

bobeasterday committed:
Return of fuzzy version matching

bobeasterday committed:
Update the readme with new domain prefix and correct plugman instructions

bobeasterday committed:
Merge pull request #61 from phonegap-build/bobeast-fixes

Finalize pre-3.0.0 plugin fixes

bobeasterday committed:
add support for 3.0.0

bobeasterday committed:
Merge pull request #49 from rjmunro/patch-1

Fix typo: successCallback should be errorCallback

bobeasterday committed:
Merge pull request #32 from florianbachmann/master

Use & instead of your large and error prone if statements

bobeasterday committed:
Bump version after some miner pull requests

smorstabilini committed:
added a missing comma after errorHanlder

bobeasterday committed:
Merge pull request #65 from smorstabilini/master

Added a missing comma after errorHandler

Commit:
Fixes Cordova error when trying to open the app from the notification bar.

Commit:
Sets default notification sound, vibration and lights and removes useless method.

bobeasterday committed:
Merge pull request #69 from rubennorte/master

Fixes Cordova error when trying to open the app from the notification bar

bobeasterday committed:
Merge pull request #64 from rjmunro/feature/eliminate-get-tasks

Eliminate need for GET_TASKS permission

armno committed:
enable syntax highlighting for code examples

armno committed:
use tabs instead of spaces for indentation (to be consistent)

madebycm committed:
Added getString method to setContentTitle and setTicker

ContentTitle and Ticker can now be set from the GCM server via the
"title" field in the "data" parameter (appName is no longer hard coded).

edewit committed:
added foreground boolean

edewit committed:
initially app is in foreground

edewit committed:
do not add notification if application in foreground

bobeasterday committed:
Don't use two different bundle ids

bobeasterday committed:
Death of a bad idea

bobeasterday committed:
Merge pull request #71 from armno/readme-syntax-highlighting

Enable syntax highlighting for code examples in readme file

bobeasterday committed:
Merge pull request #76 from madebycm/master

Added getString method to setContentTitle and setTicker

bobeasterday committed:
Merge pull request #80 from edewit/foreground

added foreground boolean

bobeasterday committed:
Fix Application badge number issue for >3.0

Commit:
getString should be extras.getString

bobeasterday committed:
Merge pull request #83 from igama/master

Android : getString should be extras.getString

mkuklis committed:
Added clobbers

mkuklis committed:
Added PushNotification to module.exports.

mkuklis committed:
Fixed clobbers

chriswiggins committed:
Encapsulating strings in single quotes is not valid JSON. Change to double quotes

chriswiggins committed:
Remove trailing comma for foreground flag and convert to double quotes

edewit committed:
call callbacks

edewit committed:
Merge branch 'master' of github.com:phonegap-build/PushPlugin into callback

bobeasterday committed:
Merge pull request #98 from edewit/callback

Callback now invoked

bobeasterday committed:
Merge pull request #91 from chriswiggins/json_fix

JSON fix

bobeasterday committed:
Merge pull request #87 from mkuklis/master

Added clobbers for PushNotification

Commit:
#85 No notifications in android when the app is closed

Allow the plugin to know when the app is no longer in foreground so that it may receive push notifications

bobeasterday committed:
Merge pull request #99 from scotthooker/master

#85 No notifications in android when the app is closed

bobeasterday committed:
bump version for cold-start notification fix

Commit:
Initially the app is not inforeground

We can rely on the the other activity functions setting foreground to be true

Commit:
remove changes

Commit:
this is the best case

We should always send a notification this is better than never sending one

bobeasterday committed:
Merge pull request #102 from scotthooker/master

#85 - new approach

bobeasterday committed:
Make notifications more consistent when not in the foreground

Commit:
json key should be surrounded by quotation

cuatl committed:
Update plugin.xml

adds android.permission.VIBRATE needed for android 2.3.x

bobeasterday committed:
Merge pull request #114 from cuatl/master

Update plugin.xml

Commit:
Allow for multiple notifications in Android
This allows for multiple notifications to be displayed on Android.  Simply add data with key "notId" and an integer value when sending via GCM.  Differentiate between different notifications by setting different notIds.  Defaults to 0 if not key supplied.

Commit:
Allow for multiple notifications in Android
This allows for multiple notifications to be displayed on Android.  Simply add data with key "notId" and an integer value when sending via GCM.  Differentiate between different notifications by setting different notIds.  Defaults to 0 if not key supplied.

Commit:
escape newlines, double-quotes and back-slashes of push message

bobeasterday committed:
On Android, don't post notification twice if in the foreground

bobeasterday committed:
Merge pull request #111 from ma2/master

json key should be surrounded by quotation

bobeasterday committed:
Capitalization error in example

cuatl committed:
fix e.regid in README

keab42 committed:
Update README.md

fesor committed:
Fixed quoute unescaping for non string values

sebastiansier committed:
Added instrucction to Android manual installation

bobeasterday committed:
Merge pull request #122 from cuatl/master

fix e.regid in README

bobeasterday committed:
Merge pull request #129 from fesor/master

Fixed quotes unescaping for non string values

bobeasterday committed:
Merge pull request #130 from sebastiansier/android_installation

Added instrucction to Android manual installation

bobeasterday committed:
Fix some assumptions about payload content.

bobeasterday committed:
Merge pull request #124 from keab42/patch-1

Update README.md

bobeasterday committed:
shutdown cleanup

mkuklis committed:
improved module condition - issue #87

Commit:
Mark PushHandlerActivity as exported (avoids security exception on
Intent dispatch)

Commit:
Mark PushHandlerActivity as exported (avoids security exception on
Intent dispatch)

EddyVerbruggen committed:
Added WP8 support

Commit:
Added amazon-fireos platform.

Commit:
Added amazon-fireos platform.

Commit:
Updated readme file with Fire OS details.

Added server script and readme for Amazon Device Messaging.

Commit:
Updated readme file with Fire OS details.

Added server script and readme for Amazon Device Messaging.

mkuklis committed:
Added support for defaults.

mkuklis committed:
Added support for defaults.

Commit:
Fixed couple of issues - 1. permission denied when launching the app from notification. 2. Message is not downloaded when resumed after hitting hoem button.

Commit:
Fixed couple of issues - 1. permission denied when launching the app from notification. 2. Message is not downloaded when resumed after hitting hoem button.

Commit:
Config options are persisted now. When app is forced stopped, we read config options from sharedPreferences.
Also, fixes error message not being shown properly.

Commit:
Config options are persisted now. When app is forced stopped, we read config options from sharedPreferences.
Also, fixes error message not being shown properly.

Commit:
Update error message for Otter1 where ADM is not supported

JIRA: https://issues.labcollab.net/browse/ATLS-1064

cr https://cr.amazon.com/r/2292352/

Commit:
Update error message for Otter1 where ADM is not supported

JIRA: https://issues.labcollab.net/browse/ATLS-1064

cr https://cr.amazon.com/r/2292352/

nadyaA committed:
Take deserialization outside the methods as it can be reused.

Adding TryCast method improves readability and decreases the code nesting.

nadyaA committed:
One channel per application and js callback functions.

It appears that MPNS opens only one channel per application and defines it by name.
Sometimes the channeluriupdate is called with delay and it looks like the channel has no Uri. Handle this by adding urichangedcallback js function.
Added js error callback to handle errors.
Added error to notify the user if they try to open new channel but already have opened one as this is throwing argument exception.
Extract the js function execution in separate method as it is reused.

nadyaA committed:
Add unregister channel

If user want to open new channel they have to have the option to close the current one.
Update the error text.
Remove the channel field as it is not necessary.

nadyaA committed:
Add unified handling of raw and toast notifications

Add API for showing toast notifications, using the system behaviour.
Expose interface for users to handle raw and toast notifications by js callbacks.
This way they can handle notifications differently than the system

nadyaA committed:
Update readme file

Align correctly tabs.

Commit:
HTML decoding the message before adding it to notification.

Commit:
HTML decoding the message before adding it to notification.

Commit:
Updated event names for the callback so existing sample code works with both android and amazon-fireos

Commit:
Updated event names for the callback so existing sample code works with both android and amazon-fireos

EddyVerbruggen committed:
typo

EddyVerbruggen committed:
Update readme file

sgrebnov committed:
Adds Windows8 support

Commit:
Merge remote-tracking branch 'upstream/master'

EddyVerbruggen committed:
Formatting fix

Commit:
[PushPlugin] Coldstart property missing from notification event in JavaScript in Cordova Push Notifications Plugin
Added logic for Coldstart support. Client JS can parse this flag.

Commit:
Updated README file. Updated index.html.
New nodeJS based server script to test amazon ADM push notification services.

russellbeattie committed:
Update README.md

Updating sound file section as e.soundname doesn't work on Fire OS. Probably should test it though.

russellbeattie committed:
Update index.html

Added in sound change

archananaik committed:
Merge pull request #2 from russellbeattie/patch-2

Update index.html

archananaik committed:
Merge pull request #1 from russellbeattie/patch-1

Update README.md

EddyVerbruggen committed:
Merge pull request #216 from archananaik/master

Updated README.md and Sample code

BBosman committed:
Fix link

EddyVerbruggen committed:
Merge pull request #217 from BBosman/incubator

Fix Cordova link: thanks for updating the link!

bmatto committed:
remove condition to test if web view is in loading state

EddyVerbruggen committed:
Merge pull request #218 from bmatto/ios-from-active-state

remove condition to test if web view is in loading state

Commit:
Fix xml validation for plugin.xml

EddyVerbruggen committed:
Merge pull request #227 from antistatique/hotfix/plugin-xml

Fix xml validation for plugin.xml - thank you!

BBosman committed:
Reference dependencies in example

BBosman committed:
Reference dependencies in example

sejas commented:
fix for iOS8 to avoid the following error:registerForRemoteNotificationTypes has been depreciated and removed from iOS 8.   It works just in xCode6

Commit:
[BlackBerry10] Add support for BB10 platform

This uses the exec proxy to wrap functionality from com.blackberry.push.

Documentation available here:
http://developer.blackberry.com/html5/documentation/v2_1/push_service.html

Commit:
Update android-support-v13.jar with the latest version

Commit:
Update android-support-v13.jar with the latest version

EddyVerbruggen committed:
https://github.com/phonegap-build/PushPlugin/pull/142 (manual patch because fork is gone)

EddyVerbruggen committed:
https://github.com/phonegap-build/PushPlugin/pull/149 (manually)

EddyVerbruggen committed:
Don't commit .project files

EddyVerbruggen committed:
[#245] partially

EddyVerbruggen committed:
Merge pull request #116 from thenewmr/master

Allow for multiple notifications in Android

EddyVerbruggen committed:
Merge pull request #142 from mkuklis/master

improved module condition - issue #87

EddyVerbruggen committed:
Merge pull request #156 from MatAtBread/master

Mark PushHandlerActivity as exported (avoids security exception)

EddyVerbruggen committed:
Merge pull request #183 from mkuklis/defaults

Added support for defaults.

EddyVerbruggen committed:
Merge pull request #244 from BBosman/plugins

Reference dependencies in example

EddyVerbruggen committed:
Merge pull request #251 from Icenium/bektchiev/update-android-support-v13

Update android-support-v13.jar with the latest version

EddyVerbruggen committed:
bump version

EddyVerbruggen committed:
Merge pull request #259 from Telerik-Verified-Plugins/master

Various (manually) merged changes

EddyVerbruggen committed:
Don't add .project files

EddyVerbruggen committed:
[#161] Clear notifications on resume

EddyVerbruggen committed:
[#161] Clear notifications on resume

EddyVerbruggen committed:
[#161] Clear notifications on resume

EddyVerbruggen committed:
Merge pull request #260 from Telerik-Verified-Plugins/master

Clear Android notifications on resume (see [#161])

sgrebnov committed:
Manual Installation for Windows8

Commit:
Make it work with iOS 8

+ removed yunnix' whitespace changes for clarity

purplecabbage committed:
Merge pull request #187 from MSOpenTech/windows8-support

Adds Windows8 support

Commit:
upped version to 2.3.0

Commit:
add ios 8 preprocessor constraints

Commit:
up version to 2.3.1

sclement41 committed:
rollback and fix to support both iOS 8 and iOS < 8, on Xcode 6 & before

sclement41 committed:
add support for iOS 7

sclement41 committed:
fix for compiling in Xcode 5

Commit:
[2.4.0] upped version

actemedia committed:
Fix WP8 first register error

FIX Problem when first register in wp8 doesn't work because callback not
initialize before call to open.

actemedia committed:
comment "Newsstand" notif type to enable badge back

Commit:
Adds windows platform support

* Adds Windows platform to config.xml
* Reuses PushPluginProxy for both windows and windows8 platforms

kenwu27 committed:
Fixed PushPluginProxy.js debug issue

Fixed the path of cordova proxy for windows and compile error in Visual Studio by adding the function wrapper

kenwu27 committed:
Removed the wrapper function

Removed the wrapper function as it will be added automatically after running a build

Commit:
deleted hardcoded support-v13 dependency

Commit:
Merge pull request #551 from kenwu27/patch-1

Fixed PushPluginProxy.js debug issue

Commit:
Merge branch 'MSOpenTech-platform-windows'

Commit:
Merge branch 'master' of https://github.com/phonegap-build/PushPlugin

Commit:
Revert "Fix WP8 first register error"

Commit:
Merge pull request #435 from actemedia/master

Fix WP8 first register error

Commit:
Merge pull request #598 from phonegap-build/revert-435-master

Revert "Fix WP8 first register error"

Commit:
[wp8] Updating docs and fixing first registeration issue

Commit:
Merge pull request #600 from phonegap-build/raghav/wp8fix

[WP8] Updating docs and fixing the first registeration issue

Commit:
Merge pull request #588 from leadermt/master

deleted hardcoded support-v13 dependency

Commit:
Updating README to add toast capable setting

macdonst committed:
Bump version 2.5.0

macdonst committed:
Merge pull request #607 from phonegap-build/raghav/toast

Updating README to add toast capable setting for windows

macdonst committed:
Deprecation notice
